contracts/AelinPool.sol

```diff
@@ -97,18 +112,61 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
         aelinDealLogicAddress = _aelinDealLogicAddress;
         aelinRewardsAddress = _aelinRewardsAddress;
 
-        if (_poolData.allowListAddresses.length > 0 || _poolData.allowListAmounts.length > 0) {
+        address[] memory tmpAllowListAddresses = _poolData.allowListAddresses;
+        uint256[] memory tmpAllowListAmounts = _poolData.allowListAmounts;
+
+        if (tmpAllowListAddresses.length > 0 || tmpAllowListAmounts.length > 0) {
             require(
-                _poolData.allowListAddresses.length == _poolData.allowListAmounts.length,
+                tmpAllowListAddresses.length == tmpAllowListAmounts.length,
                 "allowListAddresses and allowListAmounts arrays should have the same length"
             );
-            for (uint256 i = 0; i < _poolData.allowListAddresses.length; i++) {
-                allowList[_poolData.allowListAddresses[i]] = _poolData.allowListAmounts[i];
-                emit AllowlistAddress(_poolData.allowListAddresses[i], _poolData.allowListAmounts[i]);
+            for (uint256 i = 0; i < tmpAllowListAddresses.length; i++) {
+                allowList[tmpAllowListAddresses[i]] = tmpAllowListAmounts[i];
+                emit AllowlistAddress(tmpAllowListAddresses[i], tmpAllowListAmounts[i]);
             }
             hasAllowList = true;
         }
 
+        NftData[] memory tmpNftData = _poolData.nftData;
+
+        if (tmpNftData.length > 0) {
+            // if the first address supports punks or 721, the entire pool only supports 721 or punks
+            if (tmpNftData[0].collectionAddress == CRYPTO_PUNKS || NftCheck.supports721(tmpNftData[0].collectionAddress)) {
+                for (uint256 i = 0; i < tmpNftData.length; i++) {
+                    require(
+                        tmpNftData[i].collectionAddress == CRYPTO_PUNKS ||
+                            NftCheck.supports721(tmpNftData[i].collectionAddress),
+                        "can only contain 721"
+                    );
+                    nftCollectionDetails[tmpNftData[i].collectionAddress] = tmpNftData[i];
+                    emit PoolWith721(
+                        tmpNftData[i].collectionAddress,
+                        tmpNftData[i].purchaseAmount,
+                        tmpNftData[i].purchaseAmountPerToken
+                    );
+                }
+            }
+            // if the first address supports 1155, the entire pool only supports 1155
+            if (NftCheck.supports1155(tmpNftData[0].collectionAddress)) {
+                for (uint256 i = 0; i < tmpNftData.length; i++) {
+                    require(NftCheck.supports1155(tmpNftData[i].collectionAddress), "can only contain 721");
+                    nftCollectionDetails[tmpNftData[i].collectionAddress] = tmpNftData[i];
+
+                    for (uint256 j = 0; j < tmpNftData[i].tokenIds.length; j++) {
+                        nftId[tmpNftData[i].collectionAddress][tmpNftData[i].tokenIds[j]] = true;
```

Comment:
maybe instead of setting this to true you can set it to the corresponding minEligibleAmount. Not sure it matters. I will keep reading

Reply:
because we are using `nftId` in a different way for 721 and 1155, i don't think storing `minTokensEligible` to the mapping is required. 

check [this](https://github.com/cranium7811/aelin/blob/5e0426fb0e68557fda5f49de0412911a8bbed85b/contracts/AelinPool.sol#L56)